kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/model/BytesOutputRow.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Lenses.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lenses.streamreactor.connect.aws.s3.model
+
+import java.io.InputStream
+
+import io.lenses.streamreactor.connect.aws.s3.config.BytesWriteMode
+import io.lenses.streamreactor.connect.aws.s3.config.BytesWriteMode._
+
+import scala.collection.mutable.ListBuffer
+
+object BytesOutputRow {
+
+  def apply(storedByteArray: Array[Byte], bytesWriteMode: BytesWriteMode): BytesOutputRow = {
+
+    if (bytesWriteMode == KeyOnly) {
+      new BytesOutputRow(None, None, storedByteArray, Array.empty)
+
+    } else if (bytesWriteMode == ValueOnly) {
+      new BytesOutputRow(None, None, Array.empty, storedByteArray)
```

Comment:
Why Array.empty? this will send to kafka a 0 bytes array. Do we  make sure we end up creating a SourceRecord with null for the Key when we pass it to Connect?

Reply:
This is used in the sink.

If no key is expected then the key is set to empty.

Alternatively it could be an Option[Array[Byte]] but we already know not to expect a value as we have an Option for the keySize which ultimately ensures this will not be output to the file.